parser/cst.go

```diff
@@ -795,17 +806,48 @@ type With struct {
 func (w *With) Position() lexer.Position { return w.Pos }
 func (w *With) End() lexer.Position      { return shiftPosition(w.Pos, len(w.Keyword), 0) }
 
-// AliasDecl represents a declaration of an alias for a CallStmt.
-type AliasDecl struct {
-	Pos   lexer.Position
-	As    *As    `parser:"@@"`
-	Ident *Ident `parser:"@@"`
-	Func  *FuncDecl
-	Call  *CallStmt
+// BindClause represents the entire "as ..." clause on a CallStmt, with either a
+// Default side effect or a list of Binds.
+type BindClause struct {
+	Pos     lexer.Position
+	As      *As       `parser:"@@"`
+	Default *Ident    `parser:"  @@"`
+	List    *BindList `parser:"| @@"`
+	Call    *CallStmt
+}
+
+func (b *BindClause) Position() lexer.Position { return b.Pos }
+func (b *BindClause) End() lexer.Position {
+	switch {
+	case b.Default != nil:
+		return b.Default.End()
+	case b.List != nil:
+		return b.List.End()
+	default:
+		return shiftPosition(b.Pos, 1, 0)
```

Comment:
Should be panic?

Reply:
I actually just cargo-culted that line from elsewhere. Changed it to a panic though ;)